### ISSUE TITLE
cargo build: error: failed to select a version for `ibc-proto`.

### ISSUE THREAD

**Issue by taitruong** [informalsystems/hermes#3676]
I was trying to build v1.7.0 using `cargo build --release --bin hermes`, but get this error:

```sh
    Updating crates.io index
error: failed to select a version for `ibc-proto`.
    ... required by package `ibc-relayer v0.26.0 (/home/ttruong/data/development/hermes/crates/relayer)`
versions that meet the requirements `^0.38.0` (locked to 0.38.0) are: 0.38.0

the package `ibc-relayer` depends on `ibc-proto`, with features: `tonic` but `ibc-proto` does not have these features.
 It has an optional dependency with that name, but that dependency uses the "dep:" syntax in the features table, so it does not have an implicit feature with that name.
 ```
 
 I was able to build v1.6.0 though.

**Comment by romac** [comment]
This is weird, since Hermes builds fine on CI and I am also unable to reproduce this locally. 

Also this error message is plain wrong:

> the package `ibc-relayer` depends on `ibc-proto`, with features: `tonic`

since `ibc-relayer` [does not depend](https://github.com/informalsystems/hermes/blob/7e2b14ca373fd2c6746355aadc6c5e3baa1d21dd/crates/relayer/Cargo.toml#L23) on `ibc-proto` with the `tonic` feature.

Which commit are you building from?

Maybe do a `cargo clean` followed by `cargo update` and try building again?

**Comment by taitruong** [comment]
seems like it comes from `ibc-proto`: `cargo add ibc-proto --package ibc-relayer-types`

Error:
```sh
$ cargo add ibc-proto --package ibc-relayer-types
    Updating crates.io index
      Adding ibc-proto v0.38.0 to dependencies.
             Features:
             + client
             + std
             - borsh
             - json-schema
             - parity-scale-codec
             - proto-descriptor
             - serde
             - server
error: failed to select a version for `ibc-proto`.
    ... required by package `ibc-relayer v0.26.0 (/home/ttruong/data/development/hermes/crates/relayer)`
versions that meet the requirements `^0.38.0` (locked to 0.38.0) are: 0.38.
```

And yes, I did a cargo clean and working on `v1.7.0` branch.

**Comment by romac** [comment]
Ah i see then yeah the v1.7.0 tag is a bit borked. You can either install it from crates.io or build from master if that's okay with you. Sorry about that!

**Comment by taitruong** [comment]
Can you create a new release?

**Comment by taitruong** [comment]
> Ah i see then yeah the v1.7.0 tag is a bit borked. You can either install it from crates.io or build from master if that's okay with you. Sorry about that!

error when installing from crates.io:

```
$ cargo install ibc-relayer-cli
    Updating crates.io index
  Installing ibc-relayer-cli v1.7.0
    Updating crates.io index
error: failed to compile `ibc-relayer-cli v1.7.0`, intermediate artifacts can be found at `/tmp/cargo-installwWayop`

Caused by:
  failed to select a version for `ibc-proto`.
      ... required by package `ibc-chain-registry v0.26.0`
      ... which satisfies dependency `ibc-chain-registry = "^0.26.0"` of package `ibc-relayer-cli v1.7.0`
  versions that meet the requirements `^0.38.0` are: 0.38.0

  the package `ibc-chain-registry` depends on `ibc-proto`, with features: `tonic` but `ibc-proto` does not have these features.
   It has an optional dependency with that name, but that dependency uses the "dep:" syntax in the features table, so it does not have an implicit feature with that name.


  failed to select a version for `ibc-proto` which could resolve this conflict
  ```

**Comment by taitruong** [comment]
error from building on master branch:

```sh
~/data/development/hermes (master)$ cargo build --release --bin hermes
error: failed to select a version for `ibc-proto`.
    ... required by package `ibc-relayer v0.26.0 (/home/ttruong/data/development/hermes/crates/relayer)`
versions that meet the requirements `^0.38.0` (locked to 0.38.0) are: 0.38.0

the package `ibc-relayer` depends on `ibc-proto`, with features: `tonic` but `ibc-proto` does not have these features.
 It has an optional dependency with that name, but that dependency uses the "dep:" syntax in the features table, so it does not have an implicit feature with that name.


failed to select a version for `ibc-proto` which could resolve this conflict
```

**Comment by ljoss17** [comment]
Are you using Rust `v1.70.0`? If so, the issue should be solved by upgrading to Rust `v1.71.0` https://github.com/rust-lang/cargo/pull/12130

**Comment by taitruong** [comment]
> Are you using Rust `v1.70.0`? If so, the issue should be solved by upgrading to Rust `v1.71.0` [rust-lang/cargo#12130](https://github.com/rust-lang/cargo/pull/12130)

That was the problem! Upgraded to `v1.73.0` and works like a charm. Thank u!
